content/docs/configuration-services.md

```diff
@@ -0,0 +1,327 @@
+```Typescript
+export type ArithmeticsAddedServices = {
+    validation: {
+        ArithmeticsValidator: ArithmeticsValidator
+    },
+    secondGroup: {
+        AnotherServiceName: AnotherServiceType
+    },
+    nthGroup: {
+        withASubGroup: {
+             YetAnotherServiceName: YetAnotherServiceType
+        }
+    }
+}
+```
+
+Now that we have declared our new services inside of the `ArithmeticsAddedServices` type definition, we need to specify to the module how we want them to be implemented. To do so, we need to update the `ArithmeticsModule`:
+```Typescript
+export const ArithmeticsModule: Module<ArithmeticsServices, PartialLangiumServices & ArithmeticsAddedServices> = {
+    validation: {
+        ArithmeticsValidator: () => new ArithmeticsValidator()
+    }
+};
+```
+In the `ArithmeticsModule` singleton instance, we map a property with the name of our service (here `ArithmeticsValidator`) to a concrete implementation of the service. This means that every time we need to call the service named "ArithmeticsValidator", a new instance of the class `ArithmeticsValidator` will be created. 
+
+The `ArithmeticsValidator` service does not depend on other services, and no argument is passed during the instantiation of the class. If you implement a service that depends on other services, the constructor of your service should expect `<yourDslName>Services` as argument. The initializer function can expect that object as argument and pass it to your services constructor, such as:
+```Typescript
+export const ArithmeticsModule: Module<ArithmeticsServices, PartialLangiumServices & ArithmeticsAddedServices> = {
+    ServiceWithDependencies = (services) => new ServiceClass(services);
+}
+```
+The services which `ServiceClass` depends on need to be registered in the constructor:
+```Typescript
+export class ServiceClass{
+    readonly serviceOne: ServiceOne;
+    readonly serviceTwo: ServiceTwo;
+    readonly serviceN: ServiceN;
+
+    constructor(services: ArithmeticsServices){
+        this.serviceOne = services.ServiceOne;
+        this.serviceTwo = services.Group.ServiceTwo;
+        this.serviceN = services.Group.SubGroup.ServiceN;
+    }
+    /* service logic */
+}
+```
+
+### Overriding and Extending Services
+Sometimes, you would want to change the behavior of a service or add to its functionality. Thanks to the *inversion of control* via *dependency injection*, this can be achieved in one place without having to modify every piece of code that depends on the service to be overridden or extended.
+
+Let's say that you want to fully replace a service named `MyService` which implements `MyServiceInterface` interface and instantiate an instance of the class `DefaultServiceClass`.
+
+You would find the module declaration with the following line inside:
+```Typescript
+MyService: () => new DefaultServiceClass()
+```
+And `DefaultServiceClass` declared as:
+```Typescript
+export class DefaultServiceClass implements MyServiceInterface {
```

Comment:
Usually the interface has the same name as the service itself, so without the `Interface` suffix.

Reply:
Is that by design? 

I found it quite confusing that the class and interface have the exact same name when I was navigating the code.

Reply:
Well they don't have have the exact same name, the interface `MyService` describes the API other services interact with, while the implementation is usually just `DefaultMyService`, since it's a default implementation. I believe that's the common convention in most languages and frameworks, except for .Net, where it would be `IMyService` (interface) and `MyService` (implementation).